React/Modules/RCTNativeAnimationManager.m

```diff
@@ -0,0 +1,236 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTNativeAnimationManager.h"
+#import "RCTAnimation.h"
+#import <UIKit/UIKit.h>
+
+@implementation RCTNativeAnimationManager {
+  NSMutableDictionary *_animationNodes;
+  NSMutableDictionary *_animationDrivers;
+  NSMutableSet <RCTAnimationDriverNode *> *_activeAnimations;
+  NSMutableSet <RCTAnimationDriverNode *> *_finishedAnimations;
+  NSMutableSet <RCTValueAnimatedNode *> *_updatedValueNodes;
+  NSMutableSet <RCTPropsAnimatedNode *> *_propAnimationNodes;
+  CADisplayLink *displayLink_;
+}
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _animationDrivers = [NSMutableDictionary new];
+    _animationNodes = [NSMutableDictionary new];
+    _finishedAnimations = [NSMutableSet new];
+    _propAnimationNodes = [NSMutableSet new];
+    _activeAnimations = [NSMutableSet new];
+    _updatedValueNodes = [NSMutableSet new];
+  }
+  return self;
+}
+
++ (instancetype)sharedManager {
+  static RCTNativeAnimationManager *sharedManager;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    sharedManager = [[RCTNativeAnimationManager alloc] init];
+  });
+  return sharedManager;
+}
+
+- (void)createAnimatedNode:(nonnull NSNumber *)tag
+                    config:(NSDictionary *)config {
+  NSString *nodeType = config[@"type"];
+  RCTAnimatedNode *node = nil;
+  if ([nodeType isEqualToString:@"style"]) {
```

Comment:
This would be a bit more elegant (and maybe more performant) as a dictionary literal:

    static NSDictionary *map;
    static dispatch_once_t onceToken;
    dispatch_once(&onceToken, ^{
      map = @{
        @"style": [RCTStyleAnimatedNode class],
        @"value": [RCTValueAnimatedNode class],
        ...etc
      };
    });

    Class nodeClass = map[nodeType];
    if (!nodeClass) {
      RCTLogError(@"Animated node type %@ not supported natively", nodeType);
      return;
    }
    RCTAnimatedNode *node = [[nodeClass alloc] initWithTag:tag config:config];
    _animationNodes[tag] = node;